pkg/networkserver/grpc_gsns.go

```diff
@@ -1027,17 +1028,25 @@ func (ns *NetworkServer) handleJoinRequest(ctx context.Context, up *ttnpb.Uplink
 		}
 	}
 
-	js := ns.GetPeer(ctx, ttnpb.PeerInfo_JOIN_SERVER, dev.EndDeviceIdentifiers)
-	if js == nil {
-		logger.Debug("Join Server peer not found")
-		return errJoinServerNotFound
-	}
-
-	logger.Debug("Send join-request to Join Server")
-	resp, err := ttnpb.NewNsJsClient(js.Conn()).HandleJoin(ctx, req, ns.WithClusterAuth())
-	if err != nil {
-		logger.WithError(err).Warn("Join Server failed to handle join-request")
-		return err
+	var resp *ttnpb.JoinResponse
+	if js := ns.GetPeer(ctx, ttnpb.PeerInfo_JOIN_SERVER, dev.EndDeviceIdentifiers); js != nil {
+		logger.Debug("Send join-request to cluster-local Join Server")
+		resp, err = ttnpb.NewNsJsClient(js.Conn()).HandleJoin(ctx, req, ns.WithClusterAuth())
+		if err != nil {
+			logger.WithError(err).Warn("Cluster-local Join Server failed to handle join-request")
+			return err
+		}
+	} else {
+		tlsConfig, err := ns.GetTLSConfig(ctx)
```

Comment:
@htdvisser is this how this is meant to be used?

Reply:
TLS config is Join Server specific, please see https://github.com/thethingsnetwork/lorawan-stack/issues/4#issuecomment-504048663